test/integration/other/entities.js

```diff
@@ -0,0 +1,325 @@
+          .send(testData.instances.simple.one)
+          .set('Content-Type', 'application/xml')
+          .expect(200)
+          .then(() => asAlice.patch('/v1/projects/1/forms/simple/submissions/one')
+            .send({ reviewState: 'approved' })
+            .expect(200)));
+
+      await exhaust(container);
+
+      const { count } = await container.one(sql`select count(*) from entities`);
+      count.should.equal(0);
+    }));
+  });
+
+  it('should create an entity from a submission', testService(async (service, { Entities, Submissions, one }) => {
+    await service.login('alice', (asAlice) =>
+      asAlice.post('/v1/projects/1/forms?publish=true')
+        .send(testData.forms.simpleEntity)
+        .set('Content-Type', 'application/xml')
+        .expect(200)
+        .then(() => asAlice.post('/v1/projects/1/forms/simpleEntity/submissions')
+          .send(testData.instances.simpleEntity.one)
+          .set('Content-Type', 'application/xml')
+          .expect(200)));
+
+    // Look up the submission to be able to get the corresponding form def
+    const subDef = await Submissions.getCurrentDefByIds(1, 'simpleEntity', 'one', false).then((s) => s.get());
+
+    const entity = await Entities.processSubmissionDef(subDef.id);
+    entity.uuid.should.equal('12345678-1234-4123-8234-123456789abc');
+    entity.label.should.equal('Alice (88)');
+    entity.createdBy.should.equal(5); // submitter ID of Alice
+    entity.def.data.should.eql({ first_name: 'Alice', age: '88' });
+
+    const { count } = await one(sql`select count(*) from entities`);
+    count.should.equal(1);
+  }));
+
+  it('should log entity creation in audit log', testService(async (service, container) => {
+    await service.login('alice', (asAlice) =>
+      asAlice.post('/v1/projects/1/forms?publish=true')
+        .send(testData.forms.simpleEntity)
+        .set('Content-Type', 'application/xml')
+        .expect(200)
+        .then(() => asAlice.post('/v1/projects/1/forms/simpleEntity/submissions')
+          .send(testData.instances.simpleEntity.one)
+          .set('Content-Type', 'application/xml')
+          .expect(200)));
+
+    await service.login('bob', (asBob) =>
+      asBob.patch('/v1/projects/1/forms/simpleEntity/submissions/one')
+        .send({ reviewState: 'approved' })
+        .expect(200));
+
+    await exhaust(container);
+
+    const approveEvent = await container.Audits.getLatestByAction('submission.update').then((o) => o.get());
+
+    const createEvent = await container.Audits.getLatestByAction('entity.create').then((o) => o.get());
+    createEvent.actorId.should.equal(6); // Bob
+    createEvent.details.submissionId.should.equal(approveEvent.details.submissionId);
+
+    const entity = await container.Entities.getByUuid(createEvent.details.entityUuid).then((o) => o.get());
+    entity.label.should.equal('Alice (88)');
+  }));
+
+  it.skip('should log entity error in audit log', testService(async (service, container) => {
```

Comment:
After merge: revisit this test and why it is currently skipped - it throws a different kind of error than the validation errors.